### PR TITLE
Add asset type filter

### PR DESCRIPTION
**Pull request by manelmontilla**
This PR adds the capability of querying assets filtering by type to the [GET] /assets
endpoint.
For instance, a request like: ``[GET] /assets?asset_type=AWSAccount`` will return only the assets that are AWSAccounts.

### CODE REVIEW

File: graph_asset_inventory_api/openapi/graph-asset-inventory-api.yaml [hunk 1]
```diff
@@ -135,6 +135,11 @@ paths:
           schema:
             type: integer
           required: false
+        - in: query
+          name: asset_type
+          description: Type of the assets.
+          schema:
+            type: string
```
Comment (geraname) [inline graph_asset_inventory_api/openapi/graph-asset-inventory-api.yaml:142]: I believe here the `required: false` attribute is missing.

File: tests/test_api_assets.py [hunk 2]
```diff
@@ -40,6 +40,20 @@ def test_get_assets_pagination_missing_size(flask_cli, init_api_assets):
     assert compare_unsorted_list(data, init_api_assets, lambda x: x['id'])
 
 
+def test_get_assets_with_type(flask_cli, init_api_assets):
+    """Tests the API endpoint ``GET /v1/assets`` filterring by a concrete asset
```
Comment (geraname) [inline tests/test_api_assets.py]: s/filterring/filtering/

File: graph_asset_inventory_api/openapi/graph-asset-inventory-api.yaml (hunk 1 above)
Comment (manelmontilla) [reply graph_asset_inventory_api/openapi/graph-asset-inventory-api.yaml:142]: Absolutely thanks!!

File: tests/test_api_assets.py (hunk 2 above)
Comment (manelmontilla) [reply tests/test_api_assets.py]: Good catch thanks!!

**Review by geraname — APPROVED** [review]
LGTM! 👍

File: tests/test_api_assets.py [hunk 3]
```diff
@@ -40,6 +40,20 @@ def test_get_assets_pagination_missing_size(flask_cli, init_api_assets):
     assert compare_unsorted_list(data, init_api_assets, lambda x: x['id'])
 
 
+def test_get_assets_with_type(flask_cli, init_api_assets):
+    """Tests the API endpoint ``GET /v1/assets`` filtering by a concrete asset
+    type"""
```
Comment (jroimartin) [inline tests/test_api_assets.py:45]: Missing trailing dot.

File: tests/test_inventory_client.py [hunk 4]
```diff
@@ -210,6 +210,31 @@ def test_assets_pagination(cli, init_assets):
         cli.assets(0, 1000), init_assets, lambda x: x.vid)
 
 
+def test_assets_type(cli, init_assets):
+    """Tests the filter ``type`` of the method ``assets`` of the class
```
Comment (jroimartin) [inline tests/test_inventory_client.py:214]: s/type/asset_type/

File: tests/test_inventory_client.py [hunk 5]
```diff
@@ -210,6 +210,31 @@ def test_assets_pagination(cli, init_assets):
...
+    ``InventoryClient``."""
+    expected = [
+        asset for asset in init_assets if asset.asset_id.type == 'type0'
+    ]
+    assert compare_unsorted_list(
+        cli.assets(asset_type='type0'), expected, lambda x: x.vid)
+
+
+def test_assets_type_pagination(cli, init_assets):
+    """Tests the filter ``type`` and the ``pagination`` params of the method
```
Comment (jroimartin) [inline tests/test_inventory_client.py:224]: s/type/asset_type/

**Review by jroimartin — COMMENTED** [review]
@manelmontilla I know the PR is closed, but I sent a couple of comments so they can be fixed in a future PR. Given they are very minor, maybe they can be included in #5. What do you think?

**Pull request merged**